tests/Symfony/SymfonyContainerResultCacheMetaExtensionTest.php

```diff
@@ -0,0 +1,402 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Symfony;
+
+use PHPStan\Testing\PHPStanTestCase;
+use function count;
+use function file_put_contents;
+use function sys_get_temp_dir;
+use function tempnam;
+use function var_export;
+
+final class SymfonyContainerResultCacheMetaExtensionTest extends PHPStanTestCase
+{
+
+	private static string $configFilePath;
+
+	/**
+	 * This test has to check if hash of the Symfony Container is correctly calculated,
+	 * in order to do that we need to dynamically create a temporary configuration file
+	 * because `PHPStanTestCase::getContainer()` caches container under key calculated from
+	 * additional config files' paths, so we can't reuse the same config file between tests.
```

Comment:
Seems hacky and I don't understand it.

Reply:
Are you familiar with [this commit's description](https://github.com/phpstan/phpstan-symfony/pull/421/commits/a561077e30a490526498a00d5f6ac47b5a254a73)? This is done like this to work around  `PHPStanTestCase::getContainer()`, which caches the container under the hash calculated from the list of additional config files. So I can't provide these paths statically, as it would lead to cache reuse for each test case (and that would load the config pointing to one specific container XML every time, while I need different container definitions to be loaded).

Reply:
Whole I/O overhead was removed.